app/src/main/kotlin/edu/stanford/bdh/engagehf/bluetooth/BluetoothViewModel.kt

```diff
@@ -242,32 +254,27 @@ class BluetoothViewModel @Inject internal constructor(
                     )
                 )
             }
-
-            is MessagesAction.VideoSectionAction -> {
-                viewModelScope.launch {
-                    handleVideoSectionAction(messagesAction)
-                }
-            }
         }
         if (isDismissible) {
             messageRepository.completeMessage(messageId = messageId)
         }
     }
 
-    private suspend fun handleVideoSectionAction(messageAction: MessagesAction.VideoSectionAction) {
-        engageEducationRepository.getVideoBySectionAndVideoId(
+    private suspend fun handleVideoSectionAction(messageAction: MessagesAction.VideoSectionAction): Result<Video> {
+        return engageEducationRepository.getVideoBySectionAndVideoId(
             messageAction.videoSectionVideo.videoSectionId,
             messageAction.videoSectionVideo.videoId
-        ).getOrNull()?.let { video ->
-            navigator.navigateTo(
-                EducationNavigationEvent.VideoSectionClicked(
-                    video = video
-                )
+        ).onSuccess { video ->
+            navigator.navigateTo(EducationNavigationEvent.VideoSectionClicked(video))
+        }.onFailure {
+            messageNotifier.notify(
+                message = "Error while handling message action."
```

Comment:
Can we extract to string resources? Message notifier supports notify(resId) already